var/spack/repos/builtin/packages/glew/package.py

```diff
@@ -15,22 +13,16 @@ class Glew(Package):
     version('2.1.0',  sha256='04de91e7e6763039bc11940095cd9c7f880baba82196a7765f727ac05a993c95')
     version('2.0.0',  sha256='c572c30a4e64689c342ba1624130ac98936d7af90c3103f9ce12b8a0c5736764')
 
-    depends_on("cmake", type='build')
     depends_on("gl")
     depends_on('libsm')
     depends_on('libice')
 
-    def install(self, spec, prefix):
-        options = []
-        options.extend(std_cmake_args)
-
-        with working_dir('build'):
-            cmake('./cmake/', *options)
+    root_cmakelists_dir = "build/cmake"
 
-            # https://github.com/Homebrew/legacy-homebrew/issues/22025
-            # Note: This file is generated only after cmake is run
-            filter_file(r'Requires: glu',
-                        (''), '../glew.pc')
+    def cmake(self, spec, prefix):
+        super(Glew, self).cmake(spec, prefix)
 
-            make()
-            make("install")
+        # https://github.com/Homebrew/legacy-homebrew/issues/22025
+        # Note: This file is generated only after cmake is run
+        filter_file(r'Requires: glu',
+                    (''), 'glew.pc')
```

Comment:
I would do this like:
```python
@run_after('cmake')
def patch_glew_pc(self):
    filter_file('Requires: glu', '', 'glew.pc')
```

Reply:
But keep the comment, that's useful

Reply:
Will change.

BTW: What's the advantage of `run_after`? I personally consider the overriding and calling super even more readable, because it's more explicit.

Reply:
`super` is certainly more Pythonic, `run_after` is more Spack-like. I'm honestly not sure if there is an advantage, maybe @alalazo has some opinions on this?

Reply:
> I personally consider the overriding and calling super even more readable, because it's more explicit.

Having a pipeline with decorators permits to keep different operations well separated in different methods and doesn't impose any knowledge of MRO or calling convention of `super` (which should not be necessary if writing a package recipe).

Reply:
Thanks for the explanation!